contrib/init.d/pbs_mom.in

```diff
@@ -26,26 +28,28 @@ else
 fi
 
 pidof_pbs_mom() {
-   pid="-1"
-   if [ -f $PBS_HOME/mom_priv/mom.lock ];then
-   	pid=`cat $PBS_HOME/mom_priv/mom.lock`
-   fi
-   echo $pid
+	pid=`pidof pbs_mom`
```

Comment:
this is not safe to use. during a number of `pbs_mom` phases, more then one process is running that will match this query.
and nothing garantees that the pid you get here is the parent pid you are hoping to get.
so we need to store this pid somehow and read it back from a file.
```
[root@node ~]# ps axf |grep pbs_mom
 3331 pts/1    S+     0:00          \_ grep --color=auto pbs_mom
 5784 ?        SLsl   5:13 /usr/sbin/pbs_mom -H node -p -d /var/spool/pbs
14904 pts/0    Ss     0:00  \_ /usr/sbin/pbs_mom -H node -p -d /var/spool/pbs
14905 pts/0    S      0:00  |   \_ /usr/sbin/pbs_mom -H node -p -d /var/spool/pbs
[root@node ~]# pidof pbs_mom
14905 14904 5784
```
and no order from pidof, eg
```
[root@node2402 ~]# service pbs_mom restart
Restarting pbs_mom (via systemctl):                        [  OK  ]
[root@node ~]# ps axf |grep pbs_mom
 3393 pts/1    S+     0:00          \_ grep --color=auto pbs_mom
14904 pts/0    Ss     0:00 /usr/sbin/pbs_mom -H node -p -d /var/spool/pbs
14905 pts/0    S      0:00  \_ /usr/sbin/pbs_mom -H node -p -d /var/spool/pbs
 3384 ?        SLsl   0:00 /usr/sbin/pbs_mom -H node -p -d /var/spool/pbs
[root@node ~]# pidof pbs_mom
14905 14904 3384
```